File: api/src/main/java/com/aistore/module/packaginglevel/entity/PackagingLevel.java
package com.aistore.module.packaginglevel.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/** 包装层级实体（表 packaging_level） */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("packaging_level")
public class PackagingLevel {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;
    private Long skuId;
    private String levelName;
    private Integer levelSeq;
    private String unitCode;
    private BigDecimal length;
    private BigDecimal width;
    private BigDecimal height;
    private BigDecimal netWeight;
    private BigDecimal grossWeight;
    private Integer isBaseUnit;
    private Integer isSellable;
    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    private Long createdBy;
    private Long updatedBy;

    @TableLogic
    private Integer deleted;
}

File: web/src/types/packagingLevel.ts
/* 包装层级模块类型（对齐 OpenAPI 契约） */
export interface PackagingLevel {
  id: number;
  skuId: number;
  skuCode?: string | null;
  skuName?: string | null;
  levelName: string;
  levelSeq: number;
  unitCode: string;
  length?: number | null;
  width?: number | null;
  height?: number | null;
  netWeight?: number | null;
  grossWeight?: number | null;
  isBaseUnit: 0 | 1;
  isSellable: 0 | 1;
  status: 0 | 1;
  createdAt: string;
  updatedAt: string;
  createdBy?: number | null;
  updatedBy?: number | null;
}

export interface PackagingLevelSummary {
  id: number;
  skuId: number;
  skuName?: string | null;
  levelName: string;
  levelSeq: number;
  unitCode: string;
  isBaseUnit: 0 | 1;
  isSellable: 0 | 1;
  status: 0 | 1;
  createdAt: string;
}

export interface PackagingLevelListResponse {
  items: PackagingLevelSummary[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

export interface CreatePackagingLevelRequest {
  skuId: number;
  levelName: string;
  levelSeq: number;
  unitCode: string;
  length?: number;
  width?: number;
  height?: number;
  netWeight?: number;
  grossWeight?: number;
  isBaseUnit?: 0 | 1;
  isSellable?: 0 | 1;
  status?: 0 | 1;
}

export interface UpdatePackagingLevelRequest {
  levelName?: string;
  unitCode?: string;
  length?: number;
  width?: number;
  height?: number;
  netWeight?: number;
  grossWeight?: number;
  isBaseUnit?: 0 | 1;
  isSellable?: 0 | 1;
}

export interface PackagingLevelQueryParams {
  skuId?: number;
  status?: 0 | 1;
  page?: number;
  pageSize?: number;
}
